apps/src/code-studio/pd/workshop_dashboard/components/survey_results/survey_rollup_table.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {Table} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+const questionCategories = [
+  'facilitator_effectiveness',
+  'teacher_engagement',
+  'overall_success'
+];
+
+export class SurveyRollupTable extends React.Component {
+  static propTypes = {
+    rollups: PropTypes.object.isRequired,
+    questions: PropTypes.object.isRequired,
+    facilitators: PropTypes.object.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  /**
+   * Order rows that appear in roll-up table
+   * @param {Array} categories
+   * @param {Object} questions
+   * @returns {Array} Array of {key, label} objects
+   */
+  createOrderedRows(categories, questions) {
+    let orderedRows = [];
+
+    categories.forEach(category => {
+      // The order is category name first, then questions in that category
+      orderedRows.push({
+        key: category,
+        label: _.startCase(category)
+      });
+
+      let question_found = false;
+      Object.keys(questions).forEach(question_name => {
+        if (question_name.startsWith(category)) {
+          question_found = true;
+          orderedRows.push({
+            key: question_name,
+            label: questions[question_name]
+          });
+        }
+      });
+
+      // Don't keep category that doesn't have any questions
+      if (!question_found) {
+        orderedRows.pop();
+      }
+    });
+
+    return orderedRows;
+  }
+
+  createColumnLabel(facilitatorId, workshopId, courseName, facilitatorLookup) {
+    let label = '';
+
+    if (facilitatorId) {
+      let posessiveName = `${facilitatorLookup[facilitatorId]}' ${
+        _.endsWith(name, 's') ? '' : 's'
+      }`;
```

Comment:
It looks like this generates a space between the apostrophe and the s (`Brad' s`) when it should not (`Brad's`).

Reply:
ah good catch! will fix